llvm/test/Transforms/SROA/select-gep.ll

```diff
@@ -158,11 +158,20 @@ bb:
 
 define i32 @test_select_idx_memcpy(i1 %c, ptr %p) {
```

Comment:
This test looks rather odd right now. It's not at all obvious that the change is beneficial. We're still copying the same amount of data, but doing it in multiple calls of smaller chunks, and we have more GEPs.

Also we're testing both memcpy handling and gep(select(const)) now and it may be nice to separate the two.

Reply:
The transform is beneficial, though it's not entirely obvious from looking at the IR. Two of the allocas are entirely dead and will get eliminated by InstCombine, the two other memcpys will be folded to load/store, at which point this reduces to scalar code and the remaining two allocas are also removed.

The point of this test is to show that the select unfolding works correctly even if the alloca is splittable, but non-promotable.

Reply:
the two large memcpys and their corresponding allocas are now dead and will be removed later

Reply:
> The point of this test is to show that the select unfolding works correctly even if the alloca is splittable, but non-promotable.

The test does all that now. 

My point is that the test itself has `malloc` in the title, yet we're talking about `alloca` and `select` unfolding. The fact that we're doing a useful change right now makes me question what the test was supposed to test previously. IR was left essentially unchanged, so it's hard to infer the intent from that, other than memcpy is involved, somehow.  We're probably not testing that, whatever that was, now.

Id you do not think that it would make sense to test alloca spolitting vs select folding separately, then the test could use a different name, reflecting its purpose now.

Reply:
I'm confused. These are pre-commit tests specifically for this transform. They were never intended to test something else.

Reply:
Before this patch the test did not change the IR. It's not clear to me what it was intended to demonstrate, as there are no additional comments. The `memcpy` in the test name suggests that it's something about memcpy, but what exactly is it? That we never split or remove it? If that's the case, we are breaking that intent now.

Now we're also using this specific test to demonstrate that SROA does transform `gep(ptr, select(const,const))` correctly, but it's not reflected in the test name.

If someone w/o the context of this patch were to read either the old copy of this test or the one in this patch, they would not have a clear (or even, correct) idea of the purpose of the test. We should make it obvious what are the key parts we're testing here. E.g. `this test verifies that select unfolding works correctly even if the alloca is splittable, but non-promotable`, with the test name reflecting that purpose.